src/algebra/polynomial/big_operators.lean

```diff
@@ -143,6 +143,18 @@ begin
   intros x hx, simp [h x hx],
 end
 
+/--
+The degree of a product of polynomials is equal to
+the sum of the degrees, where the degree of the zero polynomial is ⊥.
+-/
+lemma degree_prod [nontrivial R] : (∏ i in s, f i).degree = ∑ i in s, (f i).degree :=
+begin
+  classical,
+  induction s using finset.induction with a s ha hs, { simp },
+  rw [prod_insert ha, sum_insert ha],
+  rw polynomial.degree_mul, rw hs,
```

Comment:
```suggestion
  induction s using finset.induction with a s ha hs,
  { simp },
  { rw [prod_insert ha, sum_insert ha],
    rw polynomial.degree_mul,
    rw hs },
```
Some suggestions regarding style:
The `{ simp }` should go on a separate line and the remaining lines should be enclosed in braces as well.
You can also combine the multiple `rw`s into a single `rw` (or an `rwa`).

Reply:
I think it's OK to dispose of a trivial base case on the same line as the `induction` or `cases` tactic. This is mentioned towards the end of [this section of the style guide](https://leanprover-community.github.io/contribute/style.html#tactic-mode).

Reply:
Sorry, was not aware of that! I was under the impression that they always had to go on a separate line.

Reply:
All of these changes are fine by me. WRT to the braces, I was just mimicking the style of `nat_degree_prod'`. I'll also note that there are a few other places in this file where the `polynomial.` is used in redundancy with the namespace.